src/senaite/core/content/sampletype.py

```diff
@@ -0,0 +1,470 @@
+            default=u"Minimum Volume"
+        ),
+        description=_(
+            u"description_sampletype_min_volume",
+            default=u"The minimum sample volume required for analysis "
+                    u"eg. '10 ml' or '1 kg'."
+        ),
+        required=True,
+    )
+
+    directives.widget(
+        "containertype",
+        UIDReferenceWidgetFactory,
+        catalog=SETUP_CATALOG,
+        query={
+            "is_active": True,
+            "sort_on": "title",
+            "sort_order": "ascending",
+        },
+    )
+    containertype = UIDReferenceField(
+        title=_(
+            u"label_sampletype_containertype",
+            default=u"Default Container Type"
+        ),
+        description=_(
+            u"description_sampletype_containertype",
+            default=u"The default container type. New sample partitions "
+                    u"are automatically assigned a container of this "
+                    u"type, unless it has been specified in more details "
+                    u"per analysis service"
+        ),
+        allowed_types=("ContainerType", ),
+        multi_valued=False,
+        required=False,
+    )
+
+    # Stickers (Data Grid)
+    directives.widget(
+        "admitted_sticker_templates",
+        DataGridWidgetFactory,
+        allow_insert=False,
+        allow_delete=False,
+        allow_reorder=False,
+        auto_append=False)
+    admitted_sticker_templates = DataGridField(
+        title=_(u"label_sampletype_admitted_stickers_templates",
+                default=u"Admitted sticker templates"),
+        description=_(u"description_sampletype_admitted_stickers_templates",
+                      default=u"Defines the stickers to use for "
+                              u"this sample type."),
+        value_type=DataGridRow(schema=IStickersRecordSchema),
+        required=True,
+        default=[{
+            'admitted': set(),
+            'small_default': None,
+            'large_default': None
+        }]
+    )
+
+
+class StickersFieldValidator(validator.SimpleFieldValidator):
+    """Custom validator for DGF
+    """
+
+    def validate(self, value):
+        if value is None:
+            value = []
+
+        _valid = False
+        _msg = _(u"Invalid sticker object format")
+
+        if len(value) == 1:
+            _valid = True
+            _errors = []
+            if not len(value[0]['admitted']):
+                _valid = False
+                _errors.append(
+                    _(u'at least one admitted sticker must be chosen'))
+            if not value[0]['small_default']:
+                _valid = False
+                _errors.append(_(u'select small default sticker'))
+            if not value[0]['large_default']:
+                _valid = False
+                _errors.append(_(u'select large default sticker'))
+            _msg = _(u"ERRORS: ") + ", ".join(_errors)
+
+        if not _valid:
+            raise Invalid(_msg)
```

Comment:
I suggest to not use underscore `_` as a prefix in function variables, it makes the code more difficult to read imho.